HelpSource/Guides/WritingUGens.schelp

```diff
@@ -2,13 +2,126 @@ title:: Writing Unit Generators
 
+The below example has only a code::.ar:: method, but you can just as easily have both code::.ar:: and code::.kr:: methods.
+
+code::
+BoringMixer {
+    *ar { arg left, right;
+        ^(left + right) * 0.5;
+    }
+}
+::
+
+Examples of pseudo-ugens found in SC include link::Classes/BLowPass4:: and link::Classes/BHiPass4::, which break down into link::Classes/SOS:: UGens.
+
+There are very few restrictions on what these classes can contain, but you should keep the following in mind:
+
+list::
+## It is courteous to leave a note in your class's help file that identifies it as a pseudo-ugen.
+## To avoid confusion, pseudo-ugens should not inherit from the base class link::Classes/UGen::.
+## A SynthDef can only have two link::Classes/LocalIn::/link::Classes/LocalOut:: pairings — one for control rate and one for audio rate. Putting LocalIn and LocalOut in a pseudo-ugen is a bad idea, as it could interfere with other feedback loops in the SynthDef.
+::
+
+section:: Basics of UGens
+
+A (real) UGen needs two components: a plugin for the server, and a class for the language. The class goes in an ordinary code::*.sc:: file, and defines the interface for your UGen in the language. This class is generally just a few lines of code that ultimately call the class method link::Classes/UGen#-multiNew::.
+
+The server plugin is where the actual UGen behavior is defined. It is given by a dynamically loaded library written in C++, whose format is platform-dependent:
+
+list::
+## code::*.so:: for *nix
+## code::*.dll:: for Windows
+## code::*.scx:: for OS X
+::
+
+A plugin file can contain more than one UGen, and it can also define things other than ugens such as buffer fill ("/b_gen") commands.
+
+When scsynth boots, it will look for plugin files in code::Platform.userExtensionDir::. Since sclang also looks for class files in the same location, the class file and the library file can go in the same place.
+
+Plug-ins are loaded during the startup of scsynth, so the server will have to be restarted after (re-)compiling a plugin. If you modify the plugin file but not the class file, you don't need to reboot the interpreter.
+
+section:: FAUST
+
+note:: This section is currently Linux only. Please expand to include instructions for other platforms. ::
+
+FAUST footnote::http://faust.grame.fr/:: is an open source DSP language that describes real-time audio units. Its greatest strength is that it can be compiled into a huge variety of contexts: standalone GUI applications, Flash plugins, Max/MSP or Pure Data externals, CSOUND opcodes, VST plugins, etc. It can also compile to SuperCollider plugins, providing the easiest way to create real UGens in SuperCollider.
```

Comment:
this sounds like `FAUST` is the preferred way to write sc ugens, which is definitely not the case. mentioning it might not be a bad idea, but it should not be recommended